src/modules/slugs/early_hooks.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+add_filter( 'qtranslate_language_detect_redirect', 'qtranxf_slugs_language_detect_redirect', 600, 3 );
+
+/**
+* Allows url redirection due to language auto detection only if site url has been requested
+*
+* @see qtranxf_check_url_maybe_redirect
+* @param string $url_lang proposed target URL for the active language to redirect to.
+* @param string $url_orig original URL supplied to browser, which needs to be standardized.
+* @param array $url_info a hash of various information parsed from original URL, cookies and other site configuration. The key names should be self-explanatory.
+*
+* @return string resulting redirection url
+*/
+function qtranxf_slugs_language_detect_redirect($url_lang, $url_orig, $url_info): string {
+   global $q_config;
+   if (site_url().'/' === $url_orig)
+       return $url_lang;
+   else {
+       if ( empty( $url_info['lang_url'] ) ){
+           return qtranxf_convertURL( $url_orig, $q_config['default_language'],false,true );
+       } else {
+           return $url_orig;
```

Comment:
There's a regression with this patch. If I set manually a URL with just the default language to force a language switch this behavior is now broken.
Example if `it` is the default language, setting `/it/` should redirect to home page after setting the cookie. With this patch I get page not found (404).

Reply:
@herrvigg it should be fixed with https://github.com/qtranslate/qtranslate-xt/pull/1364/commits/046b8c69809540f9947366255e9d5e1cf9d1e89f

Reply:
More changes with https://github.com/qtranslate/qtranslate-xt/pull/1364/commits/8c9fc822a465efb3411a89d90082e7ed4cad39e3